pennylane/transforms/qmc.py

```diff
@@ -304,25 +304,31 @@ def qmc():
         It is also possible to explore the resources required to perform the quantum Monte Carlo
         algorithm
 
-        >>> qtape = qmc.qtape.expand(depth=1)
-        >>> qml.specs(qmc)()
+        >>> qml.specs(qmc, expansion_strategy="device")()
         {'gate_sizes': defaultdict(int, {1: 15943, 2: 15812, 7: 126, 6: 1}),
-         'gate_types': defaultdict(int,
-                     {'RY': 15433,
-                      'CNOT': 15686,
-                      'Hadamard': 258,
-                      'CZ': 126,
-                      'PauliX': 252,
-                      'MultiControlledX': 126,
-                      'QFT.inv': 1}),
-         'num_operations': 31882,
-         'num_observables': 1,
-         'num_diagonalizing_gates': 0,
-         'num_used_wires': 12,
-         'depth': 30610,
-         'num_device_wires': 12,
-         'device_name': 'default.qubit.autograd',
-         'diff_method': 'backprop'}
+        'gate_types': defaultdict(int,
+                    {'RY': 7747,
+                    'CNOT': 7874,
+                    'Hadamard': 258,
+                    'CZ': 126,
+                    'Adjoint(CNOT)': 7812,
```

Comment:
Adjoint CNot should be CNOT, is it not simplified here?

Reply:
`Adjoint(CNOT)` is technically supported by the default qubit, so it doesn't get decomposed. We could technically change Line 101 to be `adjoint(fn, lazy=False)`.  Or we could consider making the default behaviour "non-lazy". But I think both solutions are outside the scope of this PR.